components/Utilities/NetworkChecker/NetworkChecker.c

```diff
@@ -0,0 +1,67 @@
+#include "NetworkChecker.h"
+
+static const char *TAG = "Utility_Wifi";
+
+/**
+ * @brief Checks the current Wi-Fi connection status.
+ * This function retrieves the information of the currently connected Wi-Fi access point.
+ * If connected, it logs the SSID; otherwise, it indicates no connection.
+ * @return bool Returns true if connected to a Wi-Fi network, false otherwise.
+ */
+bool CheckWifiStatus()
+{
+    wifi_ap_record_t ap_info;
+    esp_err_t status = esp_wifi_sta_get_ap_info(&ap_info);
+
+    if (status == ESP_OK)
+    {
+        ESP_LOGI(TAG, "Connected to SSID: %s", ap_info.ssid);
+        return true;
+    }
+    else
+    {
+        ESP_LOGE(TAG, "Not connected to any Wi-Fi");
+        return false;
+    }
+}
```

Comment:
this could be written like this:

```
bool IsWifiConnected()
{
    wifi_ap_record_t ap_info;
    esp_err_t status = esp_wifi_sta_get_ap_info(&ap_info);

    if (status != ESP_OK)
    {
        ESP_LOGE(TAG, "Not connected to any Wi-Fi");
        return false;
    }        
    ESP_LOGI(TAG, "Connected to SSID: %s", ap_info.ssid);
    return true;
}

Reply:
I fixed it, is hard to change my mindset.